osu.Game/Scoring/ScoreManager.cs

```diff
@@ -41,10 +44,11 @@ public override bool PauseImports
         }
 
         public ScoreManager(RulesetStore rulesets, Func<BeatmapManager> beatmaps, Storage storage, RealmAccess realm, IAPIProvider api,
-                            OsuConfigManager configManager = null)
+                            OsuConfigManager configManager = null, MultiplayerClient multiplayerClient = null)
```

Comment:
I vehemently disagree with roping `MultiplayerClient` into `ScoreManager`. `ScoreManager` should not know anything about multiplayer, period.

Reply:
Shouldn't `ScoreManager` have everything that could affect the score?

Reply:
Extract the piece out that it needs to know about, such as a single bool, at most. You can't just pass full classes – it creates  weird cross-component dependencies that we try to avoid.